src/NuGet.Clients/NuGet.PackageManagement.VisualStudio/VSSettings.cs

```diff
@@ -30,33 +34,46 @@ private ISettings SolutionSettings
                     ResetSolutionSettingsIfNeeded();
                 }
 
-                return NuGetUIThreadHelper.JoinableTaskFactory.Run(_solutionSettings.Item2.GetValueAsync);
+                return NuGetUIThreadHelper.JoinableTaskFactory.Run(_solutionSettings!.Item2.GetValueAsync);
             }
         }
 
-        private ISolutionManager SolutionManager { get; set; }
+        private ISolutionManager SolutionManager { get; }
+
+        private IMachineWideSettings? MachineWideSettings { get; }
 
-        private IMachineWideSettings MachineWideSettings { get; set; }
+        private IFileWatcherFactory _fileWatcherFactory;
+        private IFileWatcher _userConfigFileWatcher;
+        private IFileWatcher? _solutionConfigFileWatcher;
 
-        public event EventHandler SettingsChanged;
+        public event EventHandler? SettingsChanged;
 
         public VSSettings(ISolutionManager solutionManager)
             : this(solutionManager, machineWideSettings: null)
         {
         }
 
         [ImportingConstructor]
-        public VSSettings(ISolutionManager solutionManager, IMachineWideSettings machineWideSettings)
+        public VSSettings(ISolutionManager solutionManager, IMachineWideSettings? machineWideSettings)
+            : this(solutionManager, machineWideSettings, new FileWatcherFactory())
+        {
+        }
+
+        public VSSettings(ISolutionManager solutionManager, IMachineWideSettings? machineWideSettings, IFileWatcherFactory fileWatcherFactory)
         {
             SolutionManager = solutionManager ?? throw new ArgumentNullException(nameof(solutionManager));
             MachineWideSettings = machineWideSettings;
             SolutionManager.SolutionOpening += OnSolutionOpenedOrClosed;
             SolutionManager.SolutionClosed += OnSolutionOpenedOrClosed;
+
+            _fileWatcherFactory = fileWatcherFactory ?? throw new ArgumentNullException(nameof(fileWatcherFactory));
```

Comment:
I imagine we're ignoring the machine wide config on purpose?

Reply:
Resolving as a duplicate of https://github.com/NuGet/NuGet.Client/pull/4939#discussion_r1028711850